src/utils/unique_id.cpp

```diff
@@ -11,9 +11,22 @@
 // the public, perform publicly and display publicly, and to permit others to do so.
 //========================================================================================
 
-#include "unique_id.hpp"
+#include <algorithm>
 #include <string>
+#include <vector>
+
+#include "unique_id.hpp"
 
 namespace parthenon {
 template class UniqueIDGenerator<std::string>;
+
+std::vector<Uid_t> UidIntersection(std::vector<Uid_t> v1, std::vector<Uid_t> v2) {
+  std::vector<Uid_t> vout;
+  std::sort(v1.begin(), v1.end());
+  std::sort(v2.begin(), v2.end());
```

Comment:
Is sorting necessary here?

Reply:
`std::set_intersection` requires sorted arguments or the behavior is undefined.

Reply:
Ah ok. my bad.